lib/rex/proto/ldap/auth.rb

```diff
@@ -0,0 +1,329 @@
+            auth_info[:auth_type] = 'Anonymous'
+
+            auth_info
+          end
+        end
+
+        #
+        # Handle Simple authentication requests
+        #
+        # @param user_login [OpenStruct] User login information
+        # @param auth_info [Hash] Processed authentication information
+        #
+        # @return auth_info [Hash] Processed authentication information
+        def handle_simple_request(user_login, auth_info = {})
+          domains = []
+          names = []
+          if !user_login.name.empty?
+            if user_login.name =~ /@/
+              pub_info = user_login.name.split('@')
+              if pub_info.length <= 2
+                auth_info[:user], auth_info[:domain] = pub_info
+              else
+                auth_info[:result_code] = Net::LDAP::ResultCodeInvalidCredentials
+                auth_info[:error_msg] = "Invalid LDAP Login Attempt => DN:#{user_login.name}"
+              end
+            elsif user_login.name =~ /,/
+              begin
+                dn = Net::LDAP::DN.new(user_login.name)
+                dn.each_pair do |key, value|
+                  if key == 'cn'
+                    names << value
+                  elsif key == 'dc'
+                    domains << value
+                  end
+                end
+                auth_info[:user] = names.join('')
+                auth_info[:domain] = domains.empty? ? nil : domains.join('.')
+              rescue Net::LDAP::InvalidDNError => e
+                auth_info[:error_msg] = "Invalid LDAP Login Attempt => DN:#{user_login.name}"
+                raise e
+              end
+            elsif user_login.name =~ /\\/
+              pub_info = user_login.name.split('\\')
+              if pub_info.length <= 2
+                auth_info[:domain], auth_info[:user] = pub_info
+              else
+                auth_info[:result_code] = Net::LDAP::ResultCodeInvalidCredentials
+                auth_info[:error_msg] = "Invalid LDAP Login Attempt => DN:#{user_login.name}"
+              end
+            else
+              auth_info[:user] = user_login.name
+              auth_info[:domain] = nil
+              auth_info[:result_code] = Net::LDAP::ResultCodeInvalidCredentials
+            end
+            auth_info[:private] = user_login.authentication
+            auth_info[:private_type] = :password
+            auth_info[:result_code] = Net::LDAP::ResultCodeAuthMethodNotSupported if auth_info[:result_code].nil?
+            auth_info[:auth_type] = 'Simple'
+            auth_info
+          end
+        end
+
+        #
+        # Handle SASL authentication requests
+        #
+        # @param user_login [OpenStruct] User login information
+        # @param auth_info [Hash] Processed authentication information
+        #
+        # @return auth_info [Hash] Processed authentication information
+        def handle_sasl_request(user_login, auth_info = {})
+          if user_login.authentication[1] =~ /NTLMSSP/
```

Comment:
Should we test if this works with other SASL mechanisms, i.e. kerberos, CRAM MD5 etc.?

Reply:
That would be a valid future enhancement, currently only `NTLMSSP` has been implemented.

Reply:
I wonder if it's worth while adding an `else` statement to handle that scenario better differently in some way 🤔

Reply:
I think it/when new auth methods are added a `case` statement would be best, I could switch to a `case` with a `default` that reports unsupported, would that satisfy the desire here?